### PR TITLE
UI improvements: sparkline dedup, speedtest stats, alert deletion, reverse DNS

### PR DESCRIPTION
**Pull request by bobbyrc**
## Summary

- **Sparkline dedup**: Only push new data points when the metric value actually changes (client-side JS + server-side SQLite recording). Fixes misleading flat segments caused by the 5s SSE broadcast recording duplicate values between 60s ping measurements.
- **Speedtest min/max/avg**: Replace the download/upload sparkline canvases with computed Min, Avg, Max stat rows — more useful given the 30-minute speedtest interval.
- **Alert deletion**: Add `DELETE /api/alerts/{id}` and `DELETE /api/alerts` endpoints, plus "Clear All" button and per-row delete buttons on the alerts page with confirmation dialogs.
- **Reverse DNS hostnames**: Resolve human-readable hostnames for IP-based ping targets via `net.LookupAddr` at startup. Display hostname above the IP on ping cards when available.
- **Test suite expansion**: 35 new tests across all packages (80 → 115 total), covering previously untested engine thresholds, downtime evaluation, cooldown seeding, API validation, metric name correctness, collector edge cases, and config parse fallbacks. Also fixes a pre-existing bug in `AllCooldowns` time parsing.

## Test plan

- [x] `go vet ./...` passes
- [x] `go test -short ./...` passes (115 tests, all green)
- [x] `go build ./cmd/pingpong/` succeeds
- [ ] Manual smoke test: verify sparklines update only on value change
- [ ] Manual smoke test: verify speedtest card shows min/avg/max stats
- [ ] Manual smoke test: verify alert deletion (individual + clear all)
- [ ] Manual smoke test: verify reverse DNS hostnames appear on ping cards

🤖 Generated with [Claude Code](https://claude.com/claude-code)

### CODE REVIEW

File: internal/collector/ping.go [hunk 1]
```diff
@@ -114,3 +116,28 @@ func (p *PingCollector) ping(ctx context.Context, target string) (PingResult, er
...
+		if err != nil || len(names) == 0 {
+			continue
+		}
+		// LookupAddr returns FQDNs with trailing dot; trim it
+		hostname := strings.TrimRight(names[0], ".")
+		if hostname != "" {
+			result[target] = hostname
+		}
+	}
+	return result
```
[the remark this answers is not in the document]
Comment (bobbyrc) [reply internal/collector/ping.go]: Fixed. Switched to `net.DefaultResolver.LookupAddr(ctx, target)` with a 2-second per-lookup timeout context to prevent blocking startup on unresponsive resolvers.

File: internal/web/static/js/app.js [hunk 2]
```diff
@@ -378,13 +389,15 @@
...
-            if (sparkCanvas) {
-                drawSparkline(sparkCanvas, pingHistory[target], COLOR_GREEN);
+            var hist = pingHistory[target];
+            if (hist.length === 0 || hist[hist.length - 1] !== entry.value) {
+                pushHistory(hist, entry.value);
+                var sparkCanvas = card.querySelector('.sparkline');
+                if (sparkCanvas) {
+                    drawSparkline(sparkCanvas, hist, COLOR_GREEN);
+                }
             }
```
[the remark this answers is not in the document]
Comment (bobbyrc) [reply internal/web/static/js/app.js]: Fixed. Moved `drawSparkline()` outside the dedup guard — we always redraw the canvas on each SSE tick, but only push to the history array when the value actually changes. This prevents the blank canvas on first SSE tick after history seeding.
